### ISSUE TITLE
seeing errors starting today regarding helm: No such file or directory

### ISSUE THREAD

**Issue by sushovan23** [WyriHaximus/github-action-helm3#83]
Using kubeconfig from input
Preparing helm execution
Executing helm
/tmp/helm-exec--384-oCuLouf0s4rl-.sh: line 7: /tmp/docker-kube-config-jhrtonczhl/helm: No such file or directory
/tmp/helm-exec--384-oCuLouf0s4rl-.sh: line 7: /tmp/docker-kube-config-jhrtonczhl/helm: No such file or directory
/tmp/helm-exec--384-oCuLouf0s4rl-.sh: line 7: /tmp/docker-kube-config-jhrtonczhl/helm: No such file or directory

using 4.1 
it same action worked 100s of times before

**Comment by ashetga** [comment]
Getting the same error with v3

Existing kubeconfig found, using that and ignoring input
Preparing helm execution
Executing helm
/tmp/helm-exec--2301499-BYXh3V4neBZn-.sh: line 7: /tmp/docker-kube-config-hshghtmn/helm: No such file or directory

**Comment by WyriHaximus** [comment]
Ran it without issues two days ago. Are you on self hosted runners or github hosted runners?

**Comment by sushovan23** [comment]
Hi @WyriHaximus Yes on self hosted runners. It started working on its own after an hour n half of reporting here. Not sure what the issue was but i got multiple failure that time from different workflows and different self host runners with same error. 

It has been working as expected since then but only that 60-90mins period i saw many errors.

**Comment by laurijssen** [comment]
The same here. After switching to github-action-helm3@v3.0 it always emits  "/tmp/docker-kube-config-yljjzfoup/helm: No such file or directory" on a self-hosted runner. v2 no problems at all

**Comment by lmbaschiera** [comment]
In my case it happened out of the blue as well. We had been using it for months and all of a sudden it started failing. 

```txt
PWD: /home/runner/work/myrepo/myrepo
Existing kubeconfig found, using that and ignoring input
Preparing helm execution
Executing helm
/tmp/helm-exec--2453-otUxaWeel1kd-.sh: line 7: /tmp/docker-kube-config-ocnyywlo/helm: No such file or directory
```

**Comment by sushovan23** [comment]
funny this got updated today - yes seeing the same. 
i think its related to github outages going on right now - https://www.githubstatus.com/

**Comment by WyriHaximus** [comment]
To clarify that I understood the problem correctly for all of you:
* It happens sporadically
* Seems to align with GitHub issues

There are two things that can be done to mitigate that:
* I'll look into a retry mechanism for an X number of times and Y amount of minutes (to ensure it doesn't run for hours)
* If you have the control, install Helm and any other tools you use in your workflows. Been adding Helm and some others to mine for some time now https://github.com/WyriHaximusNet/docker-github-action-runner/blob/main/Dockerfile#L29-L33. That will save you from downloading tools every single run.

@laurijssen In your case was it just timing with a GitHub incident or does it persist?

**Comment by lmbaschiera** [comment]
Yes, in this case, the issue was caused due to get.helm.sh being unavailable, see https://github.com/helm/helm/issues/31434 - The error message (`helm: No such file or directory`) threw me off, but after inspecting the runner Dockerfile and helm get script I figured it was a Helm issue instead.
